From 2b425538ce7e6cbdaf4d949e023629c6eefecd4e Mon Sep 17 00:00:00 2001
From: Andrea Adami <andrea.adami@gmail.com>
Date: Thu, 17 Jun 2021 17:45:25 +0200
Subject: [PATCH 3/3] ofgwrite: define default WTMP_FILE for musl libc

for musl  # CONFIG_FEATURE_UTMP is not set

fix:
| busybox/libbb/messages.c:59:4: error: #error unknown path to wtmp file
|    59 | #  error unknown path to wtmp file

Signed-off-by: Andrea Adami <andrea.adami@gmail.com>
---
 busybox/libbb/messages.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/busybox/libbb/messages.c b/busybox/libbb/messages.c
index c1b7ba2..651fcff 100644
--- a/busybox/libbb/messages.c
+++ b/busybox/libbb/messages.c
@@ -50,6 +50,10 @@ const int const_int_0 = 0;
 
 #if ENABLE_FEATURE_WTMP
 /* This is usually something like "/var/adm/wtmp" or "/var/log/wtmp" */
+/* in OE musl.cfg for busybox only defines CONFIG_FEATURE_WTMP */
+#ifndef __GLIBC__
+#define WTMP_FILE "/var/log/wtmp"
+#endif
 const char bb_path_wtmp_file[] ALIGN1 =
 # if defined _PATH_WTMP
 	_PATH_WTMP;
-- 
2.17.1

